### PR TITLE
Add handling of list write to remove mutation

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#62904 Add handling of list write to remove mutation**
* #62903 Add tensor-scalar op

Differential Revision: [D30168493](https://our.internmc.facebook.com/intern/diff/D30168493)

### CODE REVIEW

File: torch/csrc/jit/passes/remove_mutation.cpp [hunk 1]
```diff
@@ -68,11 +69,27 @@ Node* MutationRemover::createSpecialMappedOp(Node* n) {
...
+bool removableSetItem(Node* n) {
+  if (n->kind() != aten::_set_item ||
+      n->input(1)->node()->kind() != prim::Constant) {
+    return false;
+  }
+  if (n->inputs().at(0)->node()->kind() != prim::ListConstruct) {
+    return false;
+  }
+  int64_t index = *constant_as<int64_t>(n->input(1));
+  if (index < 0) {
```
Comment (Krovatkin) [inline torch/csrc/jit/passes/remove_mutation.cpp:81]: nitpick: do we have `normalizeIndex` somewhere?

**Review by Krovatkin — APPROVED** [review]
:shipit:

File: torch/csrc/jit/passes/remove_mutation.cpp (hunk 1 above)
Comment (eellison) [reply torch/csrc/jit/passes/remove_mutation.cpp:81]: let me check..

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30168493).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30168493).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30168493).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30168493).

**Pull request closed without merging**